lib/parsers/heif_parser.rb

```diff
@@ -0,0 +1,430 @@
+# HEIF stands for High-Efficiency Image File format, which is basically a container that is capable of storing an image, or a sequence of images in a single file.
+# There are a number of variants of HEIF, which can be used to store images, sequences of images, or videos using different codecs.
+# The variant that Apple uses to store images and sequences of images in its iOS and macOS operating systems is High Efficiency Image Coding (HEIC), which uses HEVC / H.265 for content compression.
+class FormatParser::HEIFParser
+  include FormatParser::IOUtils
+
+  HEIF_MARKER = [0x68, 0x65, 0x69, 0x63].pack('C4') # heif marker
+  FILE_TYPE_BOX_MARKER = [0x66, 0x74, 0x79, 0x70].pack('C4') # ftyp marker
+  META_BOX_MARKER = [0x6D, 0x65, 0x74, 0x61].pack('C4') # meta marker
+  MIF1_MARKER = [0x6D, 0x69, 0x66, 0x31].pack('C4') # mif1 marker
+  MSF1_MARKER = [0x6D, 0x73, 0x66, 0x31].pack('C4') # msf1 marker
+  MEANINGLESS_BYTE = [0x00, 0x00, 0x00, 0x00].pack('C4')
+  HANDLER_MARKER = [0x68, 0x64, 0x6C, 0x72].pack('C4') # hdlr marker
+  ITEM_PROPERTIES_BOX = [0x69, 0x70, 0x72, 0x70].pack('C4') # iprp marker
+  ITEM_PROPERTIES_CONTAINER_BOX = [0x69, 0x70, 0x63, 0x6F].pack('C4') # ipco marker
+  IMAGE_SPATIAL_EXTENTS_BOX = [0x69, 0x73, 0x70, 0x65].pack('C4') # ispe marker
+  PIXEL_ASPECT_RATIO_BOX = [0x70, 0x61, 0x73, 0x70].pack('C4') # pasp marker
+  ITEM_INFO_BOX = [0x69, 0x69, 0x6E, 0x66].pack('C4') # iinf marker
+  ITEM_INFO_ENTRY = [0x69, 0x6E, 0x66, 0x65].pack('C4') # infe marker
+  MIME_MARKER = [0x6D, 0x69, 0x6D, 0x65].pack('C4') # mime marker
+  COLOUR_INFO_BOX = [0x63, 0x6F, 0x6C, 0x72].pack('C4') # colr marker
+  PIXEL_INFO_BOX = [0x70, 0x69, 0x78, 0x69].pack('C4') # pixi marker
+  RELATIVE_LOCATION_BOX = [0x72, 0x6C, 0x6F, 0x63].pack('C4') # rloc marker
+  CLEAN_APERTURE_BOX = [0x63, 0x6C, 0x61, 0x70].pack('C4') # clap marker
+  PRIMARY_ITEM_BOX = [0x70, 0x69, 0x74, 0x6D].pack('C4') # pitm marker
+  ITEM_PROPERTIES_ASSOCIATION_BOX = [0x69, 0x70, 0x6D, 0x61].pack('C4') # ipma marker
+  IMAGE_ROTATION_BOX = [0x69, 0x72, 0x6F, 0x74].pack('C4') # irot marker
+  HEADER_LENGTH = 8 # every box header has a length of 8 bytes
+  HEIC_MIME_POSSIBLE_TYPES = {
+    'heic' => :heic,
+    'heix' => :heix,
+    'heim' => :heim,
+    'heis' => :heis
+  }
+  HEIC_MIME_TYPE = 'image/heic'
+  HEIF_MIME_TYPE = 'image/heif'
+  # HEIC_SEQUENCE_MIME_TYPE = 'image/heic-sequence' # TODO: to be used when adding image-sequence parsing
```

Comment:
`# TODO:`s are usually on their own line; at least that's what the Ruby style guide recommends, mostly because code parsers are looking for them there.